include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -103,30 +103,43 @@ namespace internal
       unsigned int n_q_points = 0;
       if (use_face_values == false)
         {
-          const bool do_simplex =
+          const bool do_hex =
             std::any_of(finite_elements.begin(),
                         finite_elements.end(),
                         [](const auto &fe) {
                           for (unsigned int i = 0; i < fe->size(); ++i)
-                            if ((*fe)[i].reference_cell_type() !=
+                            if ((*fe)[i].reference_cell_type() ==
                                 ReferenceCell::get_hypercube(dim))
```

Comment:
Maybe as a follow-up, a function of the kind 
```
  ... .reference_cell().is_hyper_cube()
```
would be useful, and similarly
```
  ... .reference_cell().is_simplex()
```
I don't know how many places you use this kind of comparison in, but if it's in more than a few, then think about this as a follow-up.